retrieval/discovery/marathon/marathon.go

```diff
@@ -47,10 +48,32 @@ const appListPath string = "/v2/apps/?embed=apps.tasks"
 
 // Discovery provides service discovery based on a Marathon instance.
 type Discovery struct {
-	Servers         []string
-	RefreshInterval time.Duration
+	client          *http.Client
+	servers         []string
+	refreshInterval time.Duration
 	lastRefresh     map[string]*config.TargetGroup
-	Client          AppListClient
+	appsClient      AppListClient
+}
+
+// Initialize sets up the discovery for usage.
+func NewDiscovery(conf *config.MarathonSDConfig) (*Discovery, error) {
+	tls, err := httputil.NewTLSConfig(conf.TLSConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
```

Comment:
This is a HTTP client without a timeout. I realize it wasn't better before, but any HTTP calls should really have a timeout. See https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/#clienttimeouts